Source/WTF/wtf/generic/MainThreadGeneric.cpp

```diff
@@ -61,9 +61,7 @@ void initializeMainThreadPlatform()
 bool isMainThread()
 {
 #if HAVE(PTHREAD_MAIN_NP)
-    int mainThreadNp = pthread_main_np();
-    ASSERT(mainThreadNp != -1);
```

Comment:
Think we should leave the assert in?

Reply:
I had the same doubt, but to be on the safe side I ended up leaving this as it was before. Hopefully @darinadler would comment on this later.

(My interpretation is that maybe the original code wanted the `-1` result, which means “threads are being initialized”, to be the same as “I am the main thread, because there are no other threads yet.)

Reply:
I think it’s fine to leave the assertion in.